docs/proposal-platforms.md

```diff
@@ -235,31 +235,183 @@ suite-run functionality:
+         sure if this is deliberate) - exclude would be very bad in these cases.
+         Therefore, include is the safer option.
+
+2. Support moving some directories to different locations with symlinks to the
+   original location.
+   * New platform settings:
+     * `[symlink dir]run` (default: `none`):
+       Specifies the directory where the workflow run directories are
+       created. If specified, the workflow run directory will be created in
+       `<run dir>/<workflow-name>` and a symbolic link will be
+       created from `$HOME/cylc-run/<workflow-name>`. If not specified the
+       workflow run directory will be created in
+       `$HOME/cylc-run/<workflow-name>`. All the workflow files and the
+       `.service` directory get installed into this directory.
+     * `[symlink dir]log` (default: `none`):
+       Specifies the directory where log directories are created. If
+       specified the workflow log directory will be created in
+       `<log dir>/<workflow-name>/log` and a symbolic link will be
+       created from `$HOME/cylc-run/<workflow-name>/log`. If not specified
+       the workflow log directory will be created in
+       `$HOME/cylc-run/<workflow-name>/log`.
+     * `[symlink dir]share` (default: `none`):
+       Specifies the directory where share directories are created. If
+       specified the workflow share directory will be created in
+       `<share dir>/<workflow-name>/share` and a symbolic link will
+       be created from `<$HOME/cylc-run/<workflow-name>/share`. If not
+       specified the workflow share directory will be created in
+       `$HOME/cylc-run/<workflow-name>/share`.
+     * `[symlink dir]share/cycle` (default: `none`):
+       Specifies the directory where share/cycle directories are created.
+       If specified the workflow share/cycle directory will be created in
+       `<share/cycle dir>/<workflow-name>/share/cycle` and a symbolic link
+       will be created from `$HOME/cylc-run/<workflow-name>/share/cycle`.
+       If not specified the workflow share/cycle directory will be created in
+       `$HOME/cylc-run/<workflow-name>/share/cycle`.
+     * `[symlink dir]work` (default: `none`):
+       Specifies the directory where work directories are created. If
+       specified the workflow work directory will be created in
+       `<work dir>/<workflow-name>/work` and a symbolic link will be
+       created from `$HOME/cylc-run/<workflow-name>/work`. If not
+       specified the workflow work directory will be created in
+       `$HOME/cylc-run/<workflow-name>/work`.
+   * What happens if you want to be able to have some workflows which use
+     different directory configurations on the same "platform"? For
+     example, you might have some workflows where you want the run
+     directory on $HOME and others on $DATADIR.
+     * For platforms other than localhost the answer is to create separate
+       platforms and then choose the appropriate platform in the workflow.
```

Comment:
Not as easy as the current mechanism in Rose but hopefully this is not a common requirement.